### ISSUE TITLE
[Bug] Playback speed in Practice Mode

### ISSUE THREAD

**Issue by fuerchter** [Drewol/unnamed-sdvx-clone#440]
**Describe the bug**
Hey, I have noticed a few things about Playback speed, that I'd like to direct your attention to.
1. When changing it, the BPM in the top left does not update. As a result, the HiSpeed value won't change. Playing a 200 BPM Song with 0.5 Playback speed will scroll half as fast (tested MMod but should apply to all) as playing a 100 BPM Song. I'm assuming this is not intended?
Videos to demonstrate:
[BPM Update 200 x 0 5](https://youtu.be/TbHQU_2p4Ks)
[BPM Update 100](https://youtu.be/GFTXDtfdzrY)
I'd add a PR here, if I knew how to fix this. I think changing something for the GetBPM() calls [throughout Game.cpp](https://github.com/Drewol/unnamed-sdvx-clone/blob/2c0c39abb9c44a2db5b87df3c09204ed7e739069/Main/src/Game.cpp#L330) might do it?

2. Some FX from a chart will play at regular speed (I think) while a different speed is selected.
See [this video](https://www.youtube.com/watch?v=dBo9yB842IA) from 0:49 on for an example.

3. This is not necessarily a bug, but is there a reason, why Playback speed should be [capped at 100](https://github.com/Drewol/unnamed-sdvx-clone/blob/2c0c39abb9c44a2db5b87df3c09204ed7e739069/Main/src/PracticeModeSettingsDialog.cpp#L58)? I've tried changing that line (to 200 e.g.) and playing a song or two with it, sometimes the song just stops halfway through? If that were fixed though, why not have it as an option?

**To Reproduce**
For 1. see the videos above. The songs are from the INPUT GATE pack.
Same for 2.

**Desktop (please complete the following information):**
 - OS: Windows 7
 - GPU: GeForce GTX 580 (yes, I know...)
 - CPU: i7-2600k
 - Version: Git commit: 2021-03-21_2c0c39ab

**Comment by 123jimin** [comment]
1. It was intended but I can add an option for that.
2. Known issue but hard to fix.
3. Also known issue; it causes several problems (or even crashes) at high playback speed, so I disabled it.

**Comment by fuerchter** [comment]
Thanks for the reply.
1.: I would appreciate that, yeah. It could be "Keep original scroll speed" or something like that. To me it seems more intuitive, since other rhythm games I played (exception being Xmod in Etterna/Stepmania) do it the same way.
2. and 3.: Sorry, if this is a duplicate. I tried searching the repo for some of the terms in my original post ("Playback speed" e.g.) but didn't come up with anything.
